sdk/core/core-amqp/src/errors.ts

```diff
@@ -442,9 +442,13 @@ export enum ErrorNameConditionMapper {
  */
 export class MessagingError extends Error {
   /**
-   * @property {string} [condition] The error condition.
+   * A string label that identifies the error.
    */
-  condition?: string;
+  code?: string;
+  /**
+   * The amqp error condition.
+   */
+  amqpCondition?: string;
```

Comment:
Hmm, now I'm less sure about just logging the amqp error since we will want to log the translated error as well (that tells us if it is retryable), that would make the logs noisy since in many cases both errors would be logged to verbose mode.

I can still do that, but what about just logging condition alongside the translated error? We could also leave it out for now, there only appear to be 2 cases where 2 amqp errors map to the same code: `UnauthorizedError` and `PreconditionFailedError` and I'm not sure how useful it would be to know the specific condition in those cases.